plugins/view-transitions/includes/class-plvt-view-transition-animation.php

```diff
@@ -0,0 +1,283 @@
+
+	/**
+	 * The unique animation slug.
+	 *
+	 * @since 1.0.0
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * Unique aliases for the animation, if any.
+	 *
+	 * @since 1.0.0
+	 * @var string[]
+	 */
+	private $aliases = array();
+
+	/**
+	 * Whether the animation uses a stylesheet.
+	 *
+	 * If so, the stylesheet will be `/wp-includes/css/view-transitions-animation-{$slug}.css`.
+	 *
+	 * @since 1.0.0
+	 * @var bool
+	 */
+	private $use_stylesheet = false;
+
+	/**
+	 * Whether to apply the global view transition names while using this animation.
+	 *
+	 * @since 1.0.0
+	 * @var bool|callable
+	 */
+	private $use_global_transition_names = true;
+
+	/**
+	 * Whether to apply the post specific view transition names while using this animation.
+	 *
+	 * @since 1.0.0
+	 * @var bool|callable
+	 */
+	private $use_post_transition_names = true;
+
+	/**
+	 * Callback to get the stylesheet for the animation, as inline CSS.
+	 *
+	 * This can be used if the animation CSS requires further preparation other than simply loading its stylesheet from
+	 * the animation's corresponding CSS file.
+	 *
+	 * If the animation is configured with `$use_stylesheet = true`, the callback will receive the CSS from that file,
+	 * and the `$alias` and `$args` used as parameters. Otherwise, the callback will receive the `$alias` and `$args`
+	 * used as parameters.
+	 *
+	 * @since 1.0.0
+	 * @var callable|null
+	 */
+	private $get_stylesheet_callback = null;
+
+	/**
+	 * Default animation arguments.
+	 *
+	 * These are provided during registration, and they are used if no specific arguments are provided when using the
+	 * animation.
+	 *
+	 * @since 1.0.0
+	 * @var array<string, mixed>
+	 */
+	private $default_args = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string               $slug         Unique animation slug.
+	 * @param array<string, mixed> $config       {
+	 *     Animation configuration.
+	 *
+	 *     @type string[]      $aliases                     Unique aliases for the animation, if any. Default empty
+	 *                                                      array.
+	 *     @type bool          $use_stylesheet              Whether the animation uses a stylesheet. Default false.
+	 *     @type bool|callable $use_global_transition_names Whether to apply the global view transition names while
+	 *                                                      using this animation. Alternatively to a concrete value, a
+	 *                                                      callback can be specified to determine it dynamically.
+	 *                                                      Default true.
+	 *     @type bool|callable $use_post_transition_names   Whether to apply the post specific view transition names
+	 *                                                      while using this animation. Alternatively to a concrete
+	 *                                                      value, acallback can be specified to determine it
```

Comment:
Just one tiny typo fix:
```suggestion
	 *                                                      value, a callback can be specified to determine it
```